crates/generator/src/generator.rs

```diff
@@ -92,6 +92,60 @@ impl From<WithdrawalCellError> for Error {
     }
 }
 
+#[derive(Debug, Clone)]
+pub struct CyclesPool {
+    limit: u64,
+    available_cycles: u64,
+    syscall_config: SyscallCyclesConfig,
+}
+
+impl CyclesPool {
+    pub fn new(limit: u64, syscall_config: SyscallCyclesConfig) -> Self {
+        CyclesPool {
+            limit,
+            available_cycles: limit,
+            syscall_config,
+        }
+    }
+
+    pub fn none<'a>() -> Option<&'a mut CyclesPool> {
```

Comment:
If we use the `Option<_>` type, I think we can just remove this and pass `None` to the function.